### PR TITLE
initial work to refine Server template

### PR DESCRIPTION
**Pull request by bakedpatato**
Hi guys,

As I've mentioned before, I would like to modify the Net Core CLI template. 
In particular, I would like to add a resolver, a resolver c# type, a field added via class instead of inferred from class metadata and to add some model classes to be a more realistic use case. 

I would eventually like to go through the code generated from this template as a blog entry(dev.to? or something)  as to document the template.

Is there anything else I can do to help out on the documentation front? 

Thanks for creating this library!

### CODE REVIEW

**Comment by michaelstaib** [comment]
Sorry for the wait on this one... I will get to this next week.

**Comment by bakedpatato** [comment]
> Sorry for the wait on this one but we are completely focused in the new client API right now.
> 
> I have some concerns about complexity here. I think template is not so easy to understand any more.
> 
> I get what you want to convey to people with the auto inherit comment ... but the GetMessageBasedOnIndex is hard to follow...
> 
> The base template should help people create a GraphQL project without having to wire things up.
> 
> Think of it as `dotnet new web`.
> 
> If we want to educate people then we should do that in the StarWars example.
> 
> What are your thoughts?

Totally understand! I was thinking that I wanted to make the example a little closer to something like `dotnet new webapi`, I get how you want to leave the server template easy. 

Yeah, I wasn't pleased with how the GetMessageBasedOnIndex turned out but I couldn't figure out an easy way to convey how to make a non trivial resolver....

Would you like me to refine the starwars templates instead?

**Comment by michaelstaib** [comment]
I will close this one. I think I will have a closer look at the ms templates and then we can decide what additional templates we need.
